### PR TITLE
[WIP] Add uutils

### PR DESCRIPTION
**Pull request by aspenluxxxy**
Adds uutils, an alternative to coreutils.

**Needs to be thoroughly tested**; also needs a `coreutils-hybrid` package since `uutils` doesn't have some things such as `dd`, but I'm tired and that'll probably hurt my brain ([see this AUR package for a reference](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=coreutils-hybrid))

### CODE REVIEW

**Comment by aspenluxxxy** [comment]
List of binaries this provides, btw: 
```
-rwxr-xr-x root/root    478592 2021-03-17 20:15 ./usr/bin/arch
-rwxr-xr-x root/root    575584 2021-03-17 20:15 ./usr/bin/base32
-rwxr-xr-x root/root    575584 2021-03-17 20:15 ./usr/bin/base64
-rwxr-xr-x root/root    479744 2021-03-17 20:15 ./usr/bin/basename
-rwxr-xr-x root/root    526576 2021-03-17 20:15 ./usr/bin/cat
-rwxr-xr-x root/root    543136 2021-03-17 20:15 ./usr/bin/chgrp
-rwxr-xr-x root/root    523312 2021-03-17 20:15 ./usr/bin/chmod
-rwxr-xr-x root/root    991632 2021-03-17 20:15 ./usr/bin/chown
-rwxr-xr-x root/root    529392 2021-03-17 20:15 ./usr/bin/chroot
-rwxr-xr-x root/root    504320 2021-03-17 20:15 ./usr/bin/cksum
-rwxr-xr-x root/root    500208 2021-03-17 20:15 ./usr/bin/comm
-rwxr-xr-x root/root   1093760 2021-03-17 20:15 ./usr/bin/cp
-rwxr-xr-x root/root   1720128 2021-03-17 20:15 ./usr/bin/csplit
-rwxr-xr-x root/root    552800 2021-03-17 20:15 ./usr/bin/cut
-rwxr-xr-x root/root    996032 2021-03-17 20:15 ./usr/bin/date
-rwxr-xr-x root/root    991136 2021-03-17 20:15 ./usr/bin/df
-rwxr-xr-x root/root    573664 2021-03-17 20:15 ./usr/bin/dircolors
-rwxr-xr-x root/root    478240 2021-03-17 20:15 ./usr/bin/dirname
-rwxr-xr-x root/root    565408 2021-03-17 20:15 ./usr/bin/du
-rwxr-xr-x root/root    480800 2021-03-17 20:15 ./usr/bin/echo
-rwxr-xr-x root/root    998688 2021-03-17 20:15 ./usr/bin/env
-rwxr-xr-x root/root    525536 2021-03-17 20:15 ./usr/bin/expand
-rwxr-xr-x root/root    907600 2021-03-17 20:15 ./usr/bin/expr
-rwxr-xr-x root/root    546864 2021-03-17 20:15 ./usr/bin/factor
-rwxr-xr-x root/root    389744 2021-03-17 20:15 ./usr/bin/false
-rwxr-xr-x root/root    989408 2021-03-17 20:15 ./usr/bin/fmt
-rwxr-xr-x root/root    506496 2021-03-17 20:15 ./usr/bin/fold
-rwxr-xr-x root/root    928928 2021-03-17 20:15 ./usr/bin/groups
-rwxr-xr-x root/root   2187712 2021-03-17 20:15 ./usr/bin/hashsum
-rwxr-xr-x root/root    502320 2021-03-17 20:15 ./usr/bin/head
-rwxr-xr-x root/root    478384 2021-03-17 20:15 ./usr/bin/hostid
-rwxr-xr-x root/root    954320 2021-03-17 20:15 ./usr/bin/hostname
-rwxr-xr-x root/root    951968 2021-03-17 20:15 ./usr/bin/id
-rwxr-xr-x root/root    991296 2021-03-17 20:15 ./usr/bin/install
-rwxr-xr-x root/root    989600 2021-03-17 20:15 ./usr/bin/join
-rwxr-xr-x root/root    500144 2021-03-17 20:15 ./usr/bin/kill
-rwxr-xr-x root/root    478992 2021-03-17 20:15 ./usr/bin/link
-rwxr-xr-x root/root    970064 2021-03-17 20:15 ./usr/bin/ln
-rwxr-xr-x root/root    478496 2021-03-17 20:15 ./usr/bin/logname
-rwxr-xr-x root/root    660464 2021-03-17 20:15 ./usr/bin/ls
-rwxr-xr-x root/root    931456 2021-03-17 20:15 ./usr/bin/mkdir
-rwxr-xr-x root/root    477664 2021-03-17 20:15 ./usr/bin/mkfifo
-rwxr-xr-x root/root    459712 2021-03-17 20:15 ./usr/bin/mknod
-rwxr-xr-x root/root    980416 2021-03-17 20:15 ./usr/bin/mktemp
-rwxr-xr-x root/root    483360 2021-03-17 20:15 ./usr/bin/more
-rwxr-xr-x root/root    996128 2021-03-17 20:15 ./usr/bin/mv
-rwxr-xr-x root/root    477728 2021-03-17 20:15 ./usr/bin/nice
-rwxr-xr-x root/root   1602400 2021-03-17 20:15 ./usr/bin/nl
-rwxr-xr-x root/root    480208 2021-03-17 20:15 ./usr/bin/nohup
-rwxr-xr-x root/root    909584 2021-03-17 20:15 ./usr/bin/nproc
-rwxr-xr-x root/root    969168 2021-03-17 20:15 ./usr/bin/numfmt
-rwxr-xr-x root/root    621360 2021-03-17 20:15 ./usr/bin/od
-rwxr-xr-x root/root    506128 2021-03-17 20:15 ./usr/bin/paste
-rwxr-xr-x root/root    478800 2021-03-17 20:15 ./usr/bin/pathchk
-rwxr-xr-x root/root    523328 2021-03-17 20:15 ./usr/bin/pinky
-rwxr-xr-x root/root    927136 2021-03-17 20:15 ./usr/bin/printenv
-rwxr-xr-x root/root    525504 2021-03-17 20:15 ./usr/bin/printf
-rwxr-xr-x root/root   1775920 2021-03-17 20:15 ./usr/bin/ptx
-rwxr-xr-x root/root    909760 2021-03-17 20:15 ./usr/bin/pwd
-rwxr-xr-x root/root    929440 2021-03-17 20:15 ./usr/bin/readlink
-rwxr-xr-x root/root    931248 2021-03-17 20:15 ./usr/bin/realpath
-rwxr-xr-x root/root    481776 2021-03-17 20:15 ./usr/bin/relpath
-rwxr-xr-x root/root    971568 2021-03-17 20:15 ./usr/bin/rm
-rwxr-xr-x root/root    928240 2021-03-17 20:15 ./usr/bin/rmdir
-rwxr-xr-x root/root    968208 2021-03-17 20:15 ./usr/bin/seq
-rwxr-xr-x root/root    521328 2021-03-17 20:15 ./usr/bin/shred
-rwxr-xr-x root/root    556016 2021-03-17 20:15 ./usr/bin/shuf
-rwxr-xr-x root/root    503536 2021-03-17 20:15 ./usr/bin/sleep
-rwxr-xr-x root/root   1053984 2021-03-17 20:15 ./usr/bin/sort
-rwxr-xr-x root/root   1025328 2021-03-17 20:15 ./usr/bin/split
-rwxr-xr-x root/root    997328 2021-03-17 20:15 ./usr/bin/stat
-rwxr-xr-x root/root   3973168 2021-03-17 20:15 ./usr/bin/stdbuf
-rwxr-xr-x root/root    487312 2021-03-17 20:15 ./usr/bin/sum
-rwxr-xr-x root/root    907568 2021-03-17 20:15 ./usr/bin/sync
-rwxr-xr-x root/root    505280 2021-03-17 20:15 ./usr/bin/tac
-rwxr-xr-x root/root    951696 2021-03-17 20:15 ./usr/bin/tail
-rwxr-xr-x root/root    514064 2021-03-17 20:15 ./usr/bin/tee
-rwxr-xr-x root/root    414384 2021-03-17 20:15 ./usr/bin/test
-rwxr-xr-x root/root    552896 2021-03-17 20:15 ./usr/bin/timeout
-rwxr-xr-x root/root    947216 2021-03-17 20:15 ./usr/bin/touch
-rwxr-xr-x root/root    503008 2021-03-17 20:15 ./usr/bin/tr
-rwxr-xr-x root/root    389696 2021-03-17 20:15 ./usr/bin/true
-rwxr-xr-x root/root    931056 2021-03-17 20:15 ./usr/bin/truncate
-rwxr-xr-x root/root    526448 2021-03-17 20:15 ./usr/bin/tsort
-rwxr-xr-x root/root    476576 2021-03-17 20:15 ./usr/bin/tty
-rwxr-xr-x root/root    925440 2021-03-17 20:15 ./usr/bin/uname
-rwxr-xr-x root/root    524112 2021-03-17 20:15 ./usr/bin/unexpand
-rwxr-xr-x root/root    980688 2021-03-17 20:15 ./usr/bin/uniq
-rwxr-xr-x root/root    477424 2021-03-17 20:15 ./usr/bin/unlink
-rwxr-xr-x root/root    996768 2021-03-17 20:15 ./usr/bin/uptime
-rwxr-xr-x root/root    927248 2021-03-17 20:15 ./usr/bin/users
-rwxr-xr-x root/root    953152 2021-03-17 20:15 ./usr/bin/wc
-rwxr-xr-x root/root    522544 2021-03-17 20:15 ./usr/bin/who
-rwxr-xr-x root/root    903856 2021-03-17 20:15 ./usr/bin/whoami
-rwxr-xr-x root/root    906192 2021-03-17 20:15 ./usr/bin/yes
```

**Comment by aspenluxxxy** [comment]
2 out of 3 patches have been merged upstream now :)

**Comment by CRKatri** [comment]
Why would this be used? Does if have GNUism's or would we have to patch scripts to be POSIX?

**Comment by aspenluxxxy** [comment]
> Why would this be used?

why not

> Does if have GNUism's or would we have to patch scripts to be POSIX?

I believe it tries to replicate GNUisms - there's some places where it's not quite ready yet, hence the plans for `coreutils-hybrid`.

File: uutils.mk [hunk 1]
```diff
@@ -41,14 +86,20 @@ ifneq (,$(findstring darwin,$(MEMO_TARGET)))
 		ln -s /$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/$$(echo $$bin | rev | cut -d/ -f1 | rev) $(BUILD_DIST)/uutils/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/libexec/gnubin/$$(echo $$bin | rev | cut -d/ -f1 | rev | cut -c2-); \
 	done
 endif
-
+	
 	# uutils.mk Sign
 	$(call SIGN,uutils,general.xml)
 ifeq (,$(findstring darwin,$(MEMO_TARGET)))
+	$(LDID) -S$(BUILD_INFO)/dd.xml $(BUILD_DIST)/uutils/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/cat # Do a manual sign for dd and cat.
```
Comment (aspenluxxxy) [inline uutils.mk:93]: You sign `cat` twice.

**Comment by aspenluxxxy** [comment]
i don't care enough to continue with this anymore

if anyone else wants to continue, feel free

**Comment by Diatrus** [comment]
@CRKatri could you take this and reopen it under a branch on the main repo when you're available?

**Pull request closed without merging**